src/agent.ts

```diff
@@ -0,0 +1,98 @@
+import { Finding, HandleTransaction, TransactionEvent, FindingSeverity, FindingType } from "forta-agent";
+import {
+  DEPLOYER_ADDRESS,
+  DEPLOYMENT_TO_ADDRESS,
+  CREATE_AGENT_SIG,
+  UPDATE_AGENT_SIG,
+  DISABLE_AGENT_SIG,
+} from "./constants";
+
+const createFinding = (
+  name: string,
+  description: string,
+  alertId: string,
+  txEvent: TransactionEvent,
+  data: string
+): Finding => {
+  return Finding.fromObject({
+    name,
+    description,
+    alertId,
+    severity: FindingSeverity.Low,
+    type: FindingType.Info,
+    metadata: {
+      txHash: txEvent.hash,
+      from: txEvent.from,
+      to: txEvent.to?.toLowerCase() || "",
+      data,
+    },
+  });
+};
+
+const provideHandleTransaction = (
+  deployerAddress: string,
+  deploymentToAddress: string,
+  createAgentSig: string,
+  updateAgentSig: string,
+  disableAgentSig: string
+): HandleTransaction => {
+  return async (txEvent: TransactionEvent) => {
+    const findings: Finding[] = [];
+
+    if (
+      txEvent.from.toLowerCase() === deployerAddress.toLowerCase() &&
+      txEvent.to &&
+      txEvent.to.toLowerCase() === deploymentToAddress.toLowerCase()
+    ) {
+      const input = txEvent.transaction.data;
```

Comment:
While I can see the reasoning for this approach, it would be better for you to implement the `filterFunction()` ([source](https://docs.forta.network/en/latest/sdk/#filterfunction)) method from the Forta SDK for the reason.

This will be able to detect those function calls in the traces, whereas the bot implementation is currently solely relying on the `to` in the transaction event to be the AgentRegistry. Because of this, the bot wouldn't be able to detect a call to the AgentRegistry if it were a contract that was calling it. It would miss this: NM EOA -> contract -> AgentRegistry because the leading function selector would be a function in the first contract.

It also is a convenient tool that provides quality of life functionality for us as bot developers.